### PR TITLE
feat: implemented required question logic, user cannot move on to nex…

### PR DESCRIPTION
**Pull request by littilemike11**
…t page until all required questions are answered. Added visual to required questions and popup if required questions aren't answered

## What changed?
user cannot move to next page unless all required questions are answered

#61 
## How will this change be visible?
visual indicator for required questions and text popupwhen user hits `save` or `save & continue` while required questions are unaswered. 

Changes visible in `/src/components/AssessmentForm.tsx`
## How can you test this change?

- [ ] Automated tests
- [x] Manual tests (describe)
check if code works when user has answered all required vs has not
check if code works on pages with no required answered questions

## Notes
Suggestions on how user can be better informed on which questions are unaswered? Mentioned in code comment.

### CODE REVIEW

File: src/components/AssessmentForm.tsx [hunk 1]
```diff
@@ -57,6 +58,26 @@ const AssessmentForm: React.FC<AssessmentFormProps> = ({
...
+    // console.log(requiredQuestions)
+
+    // check if reponses are not empty, null or empty array
+    const allRequiredAnswered = requiredQuestions.every(
+      (question) =>
+        responses[question.id] != null &&
+        responses[question.id] !== "" &&
+        responses[question.id].length != 0,
+    );
+    console.log(allRequiredAnswered);
```
Comment (veryeli) [inline src/components/AssessmentForm.tsx:77]: Can you please remove all stray console.logs (including commented out ones)?

File: src/components/AssessmentForm.tsx [hunk 2]
```diff
@@ -146,17 +173,31 @@ const AssessmentForm: React.FC<AssessmentFormProps> = ({
         })}
 
         <div className="flex flex-col sm:flex-row justify-end gap-4 pt-6">
+          {/* possible alternative : instead of text at bottom of pg when required questions aren't answered, have the text pop up under the unanswered questions themself? Open to other suggestions */}
+          {requiredNotAnswered && (
+            <Text>
+              Please answer all Required <span style={{ color: "red" }}>*</span>{" "}
+              questions
```
Comment (veryeli) [inline src/components/AssessmentForm.tsx:180]: it would be more intuitive to me if 'required' was not capitalized in this sentence. And yes, I think it makes sense to add some kind of text that says 'Response required' above questions with required responses.

File: src/components/AssessmentForm.tsx [hunk 3]
```diff
@@ -146,17 +173,31 @@ const AssessmentForm: React.FC<AssessmentFormProps> = ({
...
+            </Text>
+          )}
           <button
             type="button"
             className="bg-blue-600 text-white px-5 py-2 rounded hover:bg-blue-700 transition"
-            onClick={() => onSubmit(false, isConfirmationPage)}
+            // if all required questions are answered, then submit
+            onClick={() => {
```
Comment (veryeli) [inline src/components/AssessmentForm.tsx:187]: People should not need to answer required questions to save their progress, only to submit.

**Review by veryeli — CHANGES_REQUESTED** [review]
Great start!

Please remove stray logging comments, make sure that the required logic only applies to submitting, not saving, and remove the typo with the capital R in required. I didn't investigate this but when you test, it would be good to be sure to test edge cases - for instance, when there are sub-questions.
